doc/api/esm.md

```diff
@@ -377,7 +377,34 @@ behind the `--experimental-import-meta-resolve` flag:
 * `parent` {string|URL} An optional absolute parent module URL to resolve from.
 
 > **Caveat** This feature is not available within custom loaders (it would
-> create a deadlock).
+> create a deadlock). Use [`import.meta.node.resolveURL`][] instead.
+
+### `import.meta.node.resolveURL(specifier[, parentURL])`
+
+<!--
+added: REPLACEME
+-->
+
+> Stability: 1 – Experimental
+
+* `specifier` {string|URL} The module specifier to resolve relative to the
+  `parentURL`.
+* `parentURL` {string|URL} The URL to resolve the specifier. **Default:** `import.meta.url`.
+* Returns: {Promise} Fulfills with a {URL} representing the absolute URL of the
+  resolved specifier.
+
+> **Caveat**: This is a Node.js specific API, consider using
+> [`import.meta.resolve`](#importmetaresolvespecifier) for a more portable code
+> if you don't need the second argument and do not intend your module to run in
+> a custom loader.
+
+```js
+import { readFile } from 'node:fs/promises';
+
+const data = await readFile(await import.meta.node.resolveURL('./data.txt'), 'utf-8');
+// This is equivalent to (but also available in custom loaders):
+const data2 = await readFile(new URL(import.meta.resolve('./data.txt')), 'utf-8');
```

Comment:
The comment makes it sound like the `data2` line, with `import.meta.resolve`, is available in custom loaders.

```suggestion
// This works both in application code and in custom loaders, but is specific to Node.js:
const data1 = await readFile(await import.meta.node.resolveURL('./data.txt'), 'utf-8');

// This is equivalent, and portable to other runtimes, but only works in application code
// (not in custom loaders):
const data2 = await readFile(new URL(import.meta.resolve('./data.txt')), 'utf-8');
```